roachtest: sqlsmith stream ingestion calls UpdateDeadlineMaybe on leaf txn

[(roachtest).sqlsmith/setup=seed/setting=no-mutations failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2712399&tab=buildLog) on [master@ec011620c7cf299fdbb898db692b36454defc4a2](https://github.com/cockroachdb/cockroach/commits/ec011620c7cf299fdbb898db692b36454defc4a2):

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/sqlsmith/setup=seed/setting=no-mutations/run_1
	sqlsmith.go:198,sqlsmith.go:228,test_runner.go:767: error: pq: internal error: crdb_internal.complete_stream_ingestion_job(): get-stream-ingestion-job-metadata: UpdateDeadlineMaybe() called on leaf txn
		stmt:
		WITH
			with_4711 (col_27818)
				AS (
					SELECT
						*
					FROM
						(
							VALUES
								((-8311897396447942706):::INT8),
								(8370906667990257790:::INT8),
								((-4911692604465806628):::INT8),
								(4868999840479281033:::INT8),
								((-7636973884350956732):::INT8)
						)
							AS tab_11473 (col_27818)
				)
		SELECT
			(('13:18:19.26659+12:49:00':::TIMETZ::TIMETZ + '1979-05-28':::DATE::DATE)::TIMESTAMPTZ::TIMESTAMPTZ - tab_11474._interval::INTERVAL)::TIMESTAMPTZ
				AS col_27819,
			crdb_internal.complete_stream_ingestion_job(tab_11474._int2::INT8, tab_11474._timestamptz::TIMESTAMPTZ)::INT8
				AS col_27820,
			e'\U00002603':::STRING AS col_27821,
			tab_11474._date AS col_27822
		FROM
			defaultdb.public.seed@seed__int8__float8__date_idx AS tab_11474;

```

<details><summary>More</summary><p>


Artifacts: [/sqlsmith/setup=seed/setting=no-mutations](https://teamcity.cockroachdb.com/viewLog.html?buildId=2712399&tab=artifacts#/sqlsmith/setup=seed/setting=no-mutations)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Asqlsmith%2Fsetup%3Dseed%2Fsetting%3Dno-mutations.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/ccl/streamingccl/streamingutils/utils_test.go
+++ b/pkg/ccl/streamingccl/streamingutils/utils_test.go
@@ -90,6 +90,14 @@ func TestCutoverBuiltin(t *testing.T) {
 		job.ID(), cutoverTime)
 	require.Error(t, err, "cannot cutover to a timestamp")
 
+	// Ensure that the builtin runs locally.
+	var explain string
+	err = db.QueryRowContext(ctx,
+		`EXPLAIN SELECT crdb_internal.complete_stream_ingestion_job($1, $2)`, job.ID(),
+		highWater).Scan(&explain)
+	require.NoError(t, err)
+	require.Equal(t, "distribution: local", explain)
+
 	// This should succeed since the highwatermark is equal to the cutover time.
 	var jobID int64
 	err = db.QueryRowContext(

--- a/pkg/internal/sqlsmith/schema.go
+++ b/pkg/internal/sqlsmith/schema.go
@@ -435,6 +435,12 @@ var functions = func() map[tree.FunctionClass]map[oid.Oid][]function {
 		case "pg_sleep":
 			continue
 		}
+		if strings.Contains(def.Name, "stream_ingestion") {
+			// crdb_internal.complete_stream_ingestion_job is a stateful function that
+			// requires a running stream ingestion job. Invoking this against random
+			// parameters is likely to fail and so we skip it.
+			continue
+		}
 		if strings.Contains(def.Name, "crdb_internal.force_") ||
 			strings.Contains(def.Name, "crdb_internal.unsafe_") {
 			continue

--- a/pkg/sql/sem/builtins/builtins.go
+++ b/pkg/sql/sem/builtins/builtins.go
@@ -4892,7 +4892,8 @@ may increase either contention or retry errors, or both.`,
 
 	"crdb_internal.complete_stream_ingestion_job": makeBuiltin(
 		tree.FunctionProperties{
-			Category: categoryStreamIngestion,
+			Category:         categoryStreamIngestion,
+			DistsqlBlocklist: true,
 		},
 		tree.Overload{
 			Types: tree.ArgTypes{

